components/script_traits/lib.rs

```diff
@@ -720,6 +720,8 @@ pub enum ConstellationMsg {
     IsReadyToSaveImage(HashMap<PipelineId, Epoch>),
     /// Inform the constellation of a key event.
     Keyboard(KeyboardEvent),
+    /// Whether to allow script to navigate.
+    AllowNavigation(TopLevelBrowsingContextId, PipelineId, ServoUrl, bool),
```

Comment:
Just to make things easier to understand, can you rename `AllowNavigation` to `AllowNavigationResponse`?